js/src/forum/components/PostStream.js

```diff
@@ -360,18 +357,22 @@ export default class PostStream extends Component {
     return Promise.all([$container.promise(), this.stream.loadPromise]).then(() => {
       m.redraw.sync();
 
-      // After post data has been loaded in, we will attempt to scroll back
-      // to the top of the requested post (or to the top of the page if the
-      // first post was requested). In some cases, we may have scrolled to
-      // the end of the available post range, in which case, the next range
-      // of posts will be loaded in. However, in those cases, the post we
-      // requested won't exist, so scrolling to it would cause an error.
-      // Accordingly, we start by checking that it's offset is defined.
-      const offset = $(`.PostStream-item[data-index=${index}]`).offset();
-      if (index === 0) {
+      // Rendering post contents will probably throw off our position.
+      // To counter this, we'll scroll either:
+      //   - To the reply placeholder (aligned with composer top)
+      //   - To the top of the page if we're on the first post
+      //   - To the top of a post (if that post exists)
+      // If the post does not currently exist, it's probably
+      // outside of the range we loaded in, so we won't adjust anything,
+      // as it will soon be rendered by the "load more" system.
+      let itemOffset;
+      if (reply) {
+        const $placeholder = $('.PostStream-item:last-child');
+        $(window).scrollTop($placeholder.offset().top + $placeholder.height() - $(window).height() + app.composer.computedHeight());
+      } else if (index === 0) {
         $(window).scrollTop(0);
-      } else if (offset) {
-        $(window).scrollTop($(`.PostStream-item[data-index=${index}]`).offset().top - this.getMarginTop());
+      } else if ((itemOffset = $(`.PostStream-item[data-index=${index}]`).offset())) {
```

Comment:
`index` can be undefined. If we passed in an `$item` to the `scrollToItem` function that has length zero `const index = $item.data('index');` would be undefined. In that case it won't scroll to the post after the promise resolves, and it would also in the next step set `this.stream.index = undefined` in `updateScrubberHeight`, which could result in bugs. It's not specifically something this PR has introduced, but might be worth fixing here.

Reply:
If `$item` is empty, there isn't much we can do though: we'd need to pass in the post index to `scrollToItem`, which defeats the purpose of having `scrollToItem` be its own function in the first place.

Have you run into cases where `$item` is passed in empty? `triggerScroll` is called in `onupdate`, after DOM modifications have been made, so we should expect the post item container html elements to match the range set in `PostStreamState` when we triggered the scroll.

Reply:
I think it's fine then. I just tested it, and haven't run into that case.